src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -331,6 +334,35 @@ internal virtual LexicalSortKey GetLexicalSortKey()
         /// </summary>
         public abstract ImmutableArray<Location> Locations { get; }
 
+        public virtual bool HasLocationInTree(SyntaxTree tree)
+        {
+            // Simple (but allocating) impl that can be overridden in subtypes if they show up in traces.
+            foreach (var location in Locations)
+            {
+                if (location.SourceTree == tree)
+                    return true;
+            }
+
+            return false;
+        }
+
+#nullable enable
+
+        public virtual Location? TryGetFirstLocation()
+        {
+            // Simple (but allocating) impl that can be overridden in subtypes if they show up in traces.
+            var locations = this.Locations;
+            return locations.IsEmpty ? null : locations[0];
+        }
+
+        public virtual Location GetFirstLocation()
```

Comment:
It feels like this API is redundant. At least, if we really really need a throwing helper, it shouldn't be virtual, and should be implemented in terms of ```TryGetFirstLocation```, I think.

Reply:
i would be fine with that.  i was trying to stick as closely as i could to the original semantics, but your approach is def simpler with the same end effect.  thanks!